recipe-server/client/control_new/components/recipes/PreferenceExperimentFields.js

```diff
@@ -219,17 +220,39 @@ export class ExperimentBranches extends React.Component {
   }
 }
 
+@autobind
 export class StringPreferenceField extends React.Component {
+  static propTypes = {
+    onChange: PropTypes.func.isRequired,
+  };
+
+  constructor(props) {
+    super(props);
+    this.shouldNotify = true;
+  }
+
+  handleChange(event) {
+    const { onChange } = this.props;
+    onChange(event.target.value.trim());
+  }
   render() {
+    const { onChange, ...other } = this.props;
     return (
-      <Input {...this.props} />
+      <Input
+        onChange={this.handleChange}
+        {...other}
+      />
     );
   }
 }
 
 export class BooleanPreferenceField extends React.Component {
   static propTypes = {
-    value: PropTypes.any,
+    value: PropTypes.bool,
+  }
+
+  static defaultProps = {
+    value: null,
```

Comment:
This should probably be `false`.

Reply:
I think that this is `null` initially because by default the Preference Value is neither true nor false.

![screen shot 2017-08-31 at 10 37 55 am](https://user-images.githubusercontent.com/1198628/29936859-82b6cace-8e38-11e7-9b3a-8dc2dd399b2a.png)